src/minerva_lib/blend.py

```diff
@@ -1,102 +1,87 @@
-    f_vol = f_img[:, :, np.newaxis]
-    f_bgr = np.repeat(f_vol, 3, 2) * color
-    return np.round(255 * f_bgr).astype(np.uint8)
 
+def linear_rgb(channels):
+    '''Blend all channels into one normalized image
 
-def linear_bgr(channels):
-    '''Blend all channels given
     Arguments:
         channels: List of dicts of channels to blend with rendering settings:
             {
                 image: Numpy image data
-                color: N-channel by b, g, r float32 color
-                min: Range minimum, float32 range
-                max: Range maximum, float32 range
+                color: r, g, b float32 array within 0, 1
+                min: Range minimum, float32 range 0, 1
+                max: Range maximum, float32 range 0, 1
             }
 
     Returns:
-        uint8 y by x by 3 color BGR image
+        float32 y by x by r, g, b color image within 0, 1
     '''
 
-    # Get the number of channels
-    num_channels = len(channels)
-
-    # Must be at least one channel
-    if num_channels < 1:
-        raise ValueError('At least one channel must be specified')
-
-    # Ensure that dimensions of all channels are equal
-    shape = channels[0]['image'].shape
-    for channel in channels:
-        if channel['image'].shape != shape:
-            raise ValueError('All channel images must have equal dimensions')
-
-    # Shape of 3 color image
-    shape_color = shape + (3,)
-
-    # Final buffer for blending
-    image_buffer = np.zeros(shape_color, dtype=np.float32)
+    num_channels = 0
 
-    # Process all channels
-    for channel in channels:
+    # rescaled images and normalized colors
+    for image, color in map(clip_image, channels):
 
-        image = channel['image']
-        color = channel['color']
-        min_ = channel['min']
-        max_ = channel['max']
+        num_channels += 1
+        if num_channels == 1:
 
-        # Scale the dynamic range
-        img_ranged = to_f32(image)
+            # Needs image to be 2D ndarray
+            out_shape = image.shape + (3,)
 
-        # Maximum color for this channel
-        avg_factor = 1.0 / num_channels
-        color_factor = color * avg_factor
+            # Output buffer for blending
+            out_buffer = np.zeros(out_shape, dtype=np.float32)
 
-        # Fraction of full range
-        clip_size = max_ - min_
+        # Add all three channels to output buffer
+        rgb_image = make_rgb(image, color)
+        out_buffer[:, :, 0] += next(rgb_image)
+        out_buffer[:, :, 1] += next(rgb_image)
+        out_buffer[:, :, 2] += next(rgb_image)
 
-        # Apply the range
-        img_ranged[(img_ranged < min_) | (img_ranged > max_)] = min_
-        img_norm = (img_ranged - min_) / clip_size
-
-        # Add the colored data to the image
-        y_shape, x_shape = img_norm.shape
-        img_color = f32_to_bgr(img_norm, color_factor)
-        image_buffer[0:y_shape, 0:x_shape] += img_color
+    # Must be at least one channel
```

Comment:
Remove as per previous comments

Reply:
done `refactor composite_image for single channel`